packages/plugin-core/src/commands/RefactorHierarchyV2.ts

```diff
@@ -372,19 +377,54 @@ export class RefactorHierarchyCommandV2 extends BasicCommand<
     return resp === "Proceed";
   }
 
+  prepareProxyMetricPayload(capturedNotes: DNodeProps[]) {
+    const numChildrenAcc = capturedNotes.map((note) => note.children.length);
+    const numLinksAcc = capturedNotes.map((note) => note.links.length);
+    const numCharsAcc = capturedNotes.map((note) => note.body.length);
+    const noteDepthAcc = capturedNotes.map((note) => DNodeUtils.getDepth(note));
+    const traitsAcc = capturedNotes.flatMap((note) =>
+      note.traits && note.traits.length > 0 ? note.traits : []
+    );
+    const traitsSet = new Set(traitsAcc);
+    this._proxyMetricPayload = {
+      traits: [...traitsSet],
+      // all numX props are mean value but named as plain numX
+      // since we want to use the same name across different commands for an aggregate analysis.
+      numProcessed: capturedNotes.length,
+      numChildren: _.mean(numChildrenAcc),
+      maxNumChildren: _.max(numChildrenAcc),
+      medianNumChildren: StatisticsUtils.median(numChildrenAcc),
+      stddevNumChildren: StatisticsUtils.stddev(numChildrenAcc),
+      numLinks: _.mean(numLinksAcc),
+      maxNumLinks: _.max(numLinksAcc),
+      medianNumLinks: StatisticsUtils.median(numLinksAcc),
+      stddevNumLinks: StatisticsUtils.stddev(numLinksAcc),
+      numChars: _.mean(numCharsAcc),
```

Comment:
Is this number of characters in a note?  Is this relevant for refactoring?

Reply:
I want to do a general survey of the context of refactoring. i.e. at what point / state of a note do the users feel the need to refactor?

This is more of an exploratory approach and I'm grabbing as many things as I can from `NoteProps` that I can anonymize and aggregate to a number that I can potentially use to derive the inflection point of refactor usefulness.

So in terms of relevancy, I am adding this to figure that out. My guess is that it will be hard to quantify and we wouldn't have the ability to do so without knowing the content of the note, but I'm shooting for some learning here about a general trend.